rust/datafusion/src/datasource/parquet.rs

```diff
@@ -0,0 +1,681 @@
+                                    let mut builder = BinaryBuilder::new(row_count);
+                                    for j in 0..row_count {
+                                        let slice = b[j].slice(0, b[j].len());
+                                        builder.append_string(
+                                            &String::from_utf8(slice.data().to_vec())
+                                                .unwrap(),
+                                        )?;
+                                    }
+                                    Arc::new(builder.finish())
+                                }
+                                _ => {
+                                    return Err(ExecutionError::NotImplemented(format!(
+                                        "Error reading parquet batch (column {})",
+                                        i
+                                    )));
+                                }
+                            }
+                        }
+                        ColumnReader::ByteArrayColumnReader(ref mut r) => {
+                            let mut b: Vec<ByteArray> =
+                                Vec::with_capacity(self.batch_size);
+                            for _ in 0..self.batch_size {
+                                b.push(ByteArray::default());
+                            }
+                            match r.read_batch(self.batch_size, None, None, &mut b) {
+                                Ok((count, _)) => {
+                                    row_count = count;
+                                    let mut builder = BinaryBuilder::new(row_count);
+                                    for j in 0..row_count {
+                                        let slice = b[j].slice(0, b[j].len());
+                                        builder.append_string(
+                                            &String::from_utf8(slice.data().to_vec())
+                                                .unwrap(),
+                                        )?;
+                                    }
+                                    Arc::new(builder.finish())
+                                }
+                                _ => {
+                                    return Err(ExecutionError::NotImplemented(format!(
+                                        "Error reading parquet batch (column {})",
+                                        i
+                                    )));
+                                }
+                            }
+                        }
+                    };
+
+                    batch.push(array);
+                }
+
+                if row_count == 0 {
+                    Ok(None)
+                } else {
+                    Ok(Some(RecordBatch::try_new(self.schema.clone(), batch)?))
+                }
+            }
+            _ => Ok(None),
+        }
+    }
+}
+
+fn to_arrow(t: &Type) -> Result<Field> {
```

Comment:
Maybe you can use the schema converter in parquet::reader::schema::parquet_to_arrow_schema?

Reply:
done